docs/reference/api/storage/LittleFileSystem.md

```diff
@@ -1,11 +1,130 @@
 ## LittleFileSystem
 
-[Add description here.]
+The LittleFileSystem (littlefs) is a little fail-safe file system designed for embedded systems, specifically for microcontrollers that use flash storage.
+
+```
+   | | |     .---._____
+  .-----.   |          |
+--|o    |---| littlefs |
+--|     |---|          |
+  '-----'   '----------'
+   | | |
+```
+
+Microcontrollers and flash storage present three challenges for embedded storage: [power loss](#power-loss-resilience), [wear](#wear-leveling) and [limited RAM and ROM](#bounded-ram-and-rom). The LittleFileSystem provides a solution to all three of these problems.
+
+### Power loss resilience
+
+Embedded systems are usually without a shutdown routine and the notable lack of a user interface for recovery. With a file system that is not resilient to power loss, you rely on luck to not end up with a corrupted file system. The combination of persistent storage and unpredictable power loss creates bugs that are difficult to notice and that ruin the experience of unlucky users.
```

Comment:
Having/or not proper (firmware)shutdown routine doesn't help with power loss. So, this sentence seems irrelevant.

Reply:
Although it does increase the chance of power-loss, since it's the only shutdown option available.

I was trying to indentify why power-loss resilience is so important for embedded systems. I'm open to changing this sentence if you can think of a better wording